### DISCUSSION TITLE
Where to put pre-trained models? And the difference of a Controlnet model and a pre-trained model?

### DISCUSSION THREAD

**Discussion by sk8erchen** [Mikubill/sd-webui-controlnet#686] [Q&A]
Hi, I've gone throught the installation doc in the Code page, I'm confused there seems are two kinds of models, one is controlnet model where need to be places under models\ControlNet, the other one is called pre-trained model, I wonder what this pre-trained model is and for, and where should I put it? Thanks!

![image](https://user-images.githubusercontent.com/75480055/229169349-6189eb73-7754-437a-bd33-8949fc294491.png)

**Comment by ljleb** [comment]
The pretrained models are just another name for controlnet models in this case. The link definitely redirects to the originally shared controlnet models by Illyasviel.

You can either put difference models or full models in your models/controlnet directory and the extension will be able to load them appropriately.

**Comment by marcsyp** [comment]
Is there a way to set a different Controlnet models directory to point to a network location??

Thanks,
Marc

**Comment by marcsyp** [comment]
I think I found it?

def preload(parser):
    parser.add_argument(
        "--controlnet-dir",
        type=str,
        help="Path to directory with ControlNet models",
        default=None,
    )

I can use this flag in the startup to define my controlnet Models dir?  Going to test right now.  :)
